sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -0,0 +1,308 @@
+        recenter : str or None
+            String to make object-centered maps, such as Moon/Sun/Planet centered maps.
+            Look at sotodlib.mapmaking.parse_recentering for details.
+        tile_shape : list, optional
+            List with the shape of the tiles when using tiled maps from pixell
+        tiled : Bool, optional
+            If True, use tiled maps from pixell. If False, enmaps will be used.
+        Nsplits : int, optional
+            Number of splits that you will map simultaneously. By default is 1 when no
+            splits are requested.
+        singlestream : Bool, optional
+            If True, do not perform demodulated filter+bin mapmaking but rather regular 
+            filter+bin mapmaking, i.e. map from obs.signal rather than from obs.dsT, 
+            obs.demodQ, obs.demodU
+        
+        """
+        DemodSignal.__init__(self, name, ofmt, output, ext)
+        self.comm  = comm
+        self.comps = comps
+        self.sys   = sys
+        self.recenter = recenter
+        self.dtype = dtype
+        self.tiled = tiled
+        self.data  = {}
+        self.Nsplits = Nsplits
+        self.singlestream = singlestream
+        ncomp      = len(comps)
+        shape      = tuple(shape[-2:])
+        if tiled:
+            geo = tilemap.geometry(shape, wcs, tile_shape=tile_shape)
+            self.rhs = tilemap.zeros(geo.copy(pre=(Nsplits,ncomp,)),      dtype=dtype)
+            self.div = tilemap.zeros(geo.copy(pre=(Nsplits,ncomp,ncomp)), dtype=dtype)
+            self.hits= tilemap.zeros(geo.copy(pre=(Nsplits,)),            dtype=dtype)
+        else:
+            self.rhs = enmap.zeros((Nsplits, ncomp)     +shape, wcs, dtype=dtype)
+            self.div = enmap.zeros((Nsplits,ncomp,ncomp)+shape, wcs, dtype=dtype)
+            self.hits= enmap.zeros((Nsplits,)+shape, wcs, dtype=dtype)
+
+    def add_obs(self, id, obs, nmat, Nd, pmap=None, split_labels=None):
+        # Nd will have 3 components, corresponding to ds_T, demodQ, demodU with the noise model applied
+        """Add and process an observation, building the pointing matrix
+        and our part of the RHS. "obs" should be an Observation axis manager,
+        nmat a noise model, representing the inverse noise covariance matrix,
+        and Nd the result of applying the noise model to the detector time-ordered data.
+        """
+        ctime  = obs.timestamps
+        for n_split in range(self.Nsplits):
+            if pmap is None:
+                # Build the local geometry and pointing matrix for this observation
+                if self.recenter:
+                    rot = recentering_to_quat_lonlat(*evaluate_recentering(self.recenter, ctime=ctime[len(ctime)//2], geom=(self.rhs.shape, self.rhs.wcs), site=unarr(obs.site)))
+                else: rot = None
+                if split_labels is None:
+                    # this is the case with no splits
+                    rangesmatrix = obs.flags.glitch_flags
```

Comment:
Two comments

- I am not a fun of hard-coded 'glitch_flags', as we might want to discard turnarounds or single subscan in scans. So please make it given as an argument
- I do not want to give a name of 'rangesmatrix' to 'cuts'. Why don't you use 'cuts' for 'cuts'?

Reply:
The hard coded glitch_flags is something we will change soon. So the entire mapmaking module has this issue of needing a hard-coded flags because it was created like that several years ago. This was carried over from the way it was created from a sotodlib database of ACT tods. We have talked about this with @msilvafe, because we need to do a retroactive general review of the mapmaking module to add documentation and change several things like this. So bottom line, we are aware of a number of issues that have been part of the mapmaking module for years and we will do a PR to do a retroactive review of it.

Reply:
I changed the name of the variable to cuts also